metldata/submission_registry/submission_store.py

```diff
@@ -112,11 +97,10 @@ def insert_new(self, *, submission: models.SubmissionCreation) -> models.Submiss
         return submission_with_id
 
     def update_existing(self, *, submission: models.Submission) -> None:
-        """Update an existing submission.
+        """Update an existing submission."""
 
-        Raises:
-            SubmissionDoesNotExistError: Raised when the submission does not exist.
-        """
+        existing_submission = self.get_by_id(submission_id=submission.id)
+        if submission.id != existing_submission.id:
+            raise ValueError("The submission id cannot be changed.")
```

Comment:
Should never happen, or only when there is an inconsistency in the store between path and content.

Reply:
A sorry that was my mistake of resulting from a merge conflict. Was already solved before.